### DISCUSSION TITLE
Unable to enroll to courses

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Lern/Community#143] [Issues]
When I am trying to enroll for a course I get the following response 
API - `/learner/course/v1/enrol`
Request -
`{"request":{"courseId":"do_1138466520964055041213","userId":"9a264762-f266-484f-9c43-3c23071b73b6","batchId":"0138467016066498560"}}
`

Response - 
```
{
    "id": "api.course.enroll",
    "ver": "v1",
    "ts": "2023-07-25 10:09:29:300+0000",
    "params": {
        "resmsgid": null,
        "msgid": "6b63e2eb-0be6-1456-b184-6a5a4938bae1",
        "err": "USER_ALREADY_ENROLLED_COURSE",
        "status": "USER_ALREADY_ENROLLED_COURSE",
        "errmsg": "User has already Enrolled this course ."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}
```
Although the user is not enrolled even then we are getting this response.
Can anyone please tell me what is going wrong?

cc : @coolbung @komalm @Renuka-S @vinod303 @paritshivani @vishalchincholi1 @reshmi-nair

**Comment by vinod303** [comment]
@Xitija  can you please use try user enrolment list api and see whether there is any entry exists there for that course/batch ?  
/api/course/private/v1/user/enrollment/list/9a264762-f266-484f-9c43-3c23071b73b6

**Comment by Jayaprakash8887** [comment]
@Xitija Please let us know if this is resolved

**Reply by Xitija** [reply]
@vishalchincholi1 can you please test anc clarify this so we can close it ?

**Reply by Jayaprakash8887** [reply]
@Xitija Please update

**Reply by Xitija** [reply]
We are still facing 400 error on this API , we have asked devops to check it

**Reply by AmiableAnil** [reply]
@Xitija Please let us know if this is resolved or do you still need some help.

**Reply by Xitija** [reply]
I have moved from this task and as it is no longer relevant you can close it
